crates/hir-def/src/nameres/diagnostics.rs

```diff
@@ -22,7 +22,7 @@ pub enum DefDiagnosticKind {
 
     UnresolvedImport { id: ItemTreeId<item_tree::Import>, index: Idx<ast::UseTree> },
 
-    UnconfiguredCode { ast: AstId<ast::Item>, cfg: CfgExpr, opts: CfgOptions },
+    UnconfiguredCode { ast: AstId<AnyHasAttrs>, cfg: CfgExpr, opts: CfgOptions },
```

Comment:
I could also use something like `InFile<SyntaxNodePtr>` but wasn't sure whether it is okay to use a lower level construct here in HIR level. `AnyHasAttrs` does the trick but I had to create lots of `From` implementation for it. I wonder if there is a better way to do that

Reply:
`AnyHasAttrs` seems like the correct choice, technically anything with attributes can be unconfigured